addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java

```diff
@@ -0,0 +1,272 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.text.MessageFormat;
+import java.util.Properties;
+import java.util.UUID;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.magentatv.internal.MagentaTVControl;
+import org.openhab.binding.magentatv.internal.MagentaTVException;
+import org.openhab.binding.magentatv.internal.MagentaTVLogger;
+
+/**
+ * The {@link MagentaTVOAuth} class implements the OAuth authentication, which
+ * is used to query the userID from the Telekom platform.
+ *
+ * @author Mathias Gisch - Initial contribution - bash script (Mat_Ias)
+ * @author Markus Michels - Initial contribution (markus7017)
+ */
+public class MagentaTVOAuth {
+    private final MagentaTVLogger logger = new MagentaTVLogger(MagentaTVOAuth.class, "OAuth");
+
+    private final String OAUTH_GET_CRED_URL = "https://slbedmfk11100.prod.sngtv.t-online.de";
+    private final String OAUTH_GET_CRED_PORT = "33428";
+    private final String OAUTH_GET_CRED_URI = "/EDS/JSON/Login?UserID=Guest";
+    private final String OAUTH_USER_AGENT = "Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Mobile/14G60 (400962928)";
+
+    // OAuth autentication results
+    private String oAuthScope = "";
+    private String oAuthService = "";
+    private String oAuthClientSecret = "";
+    private String oAuthClientId = "";
+    private String accessToken = "";
+    private String epgurl = "";
+    private String epghttpsurl = "";
+
+    public String getOAuthCredentials(String accountName, String accountPassword) throws Exception {
+        logger.info("Authenticate with account '{}'", accountName);
+        Properties httpHeader = initHttpHeader();
+        String url = "";
+        String httpResponse = "";
+        String postData = "";
+        InputStream dataStream = null;
+        JsonReader reader;
+        JsonObject json;
+
+        if (accountName.isEmpty() || accountPassword.isEmpty()) {
+            throw new MagentaTVException("Credentials for OAuth missing, check thing config!");
+        }
+
+        try {
+            httpHeader = initHttpHeader();
+            url = OAUTH_GET_CRED_URL + ":" + OAUTH_GET_CRED_PORT + OAUTH_GET_CRED_URI;
+            httpHeader.setProperty(HEADER_HOST, StringUtils.substringAfterLast(OAUTH_GET_CRED_URL, "/"));
+            logger.trace("Get OAuth credentials from '{}'", url);
+            httpResponse = HttpUtil.executeUrl(HTTP_GET, url, httpHeader, null, null, NETWORK_TIMEOUT);
+
+            // Telekom uses a OAuth-based authentication to access the EPG prtoal. The
```

Comment:
typo. I also think that this block of comment should go into the class description, so that it can be seen by a javadoc generator. At the moment it is hidden in here.

Reply:
done